### PR TITLE
Relicense layout definitions under CC0

### PR DESCRIPTION
**Pull request by Julow**
Layout definitions are licensed differently from the rest of the application source code to allow use in other projects related or unrelated to Unexpected Keyboard.

Layout authors can ask to add an exception for the layout they authored. I'm not asking for everyone to answer and I reserve the right to relicense individual layouts back to GPLv3 in the future.

Note: I moved the layout definitions into the directory `srcs/layouts` in ad7314a as a general improvement and to make this change easier.

cc Interested users @Lixquid
cc Layout authors @rafasaurus @lyubomirv @Shareef101 @ElucGeek @shmVirus @GoRaN909 @mostafaayesh @lrvideckis @K4zoku @Raj9039852537 @lpv11 @iamrasel @RamKromberg @Validbit @Dimethylebutane @thabubble @eandersons @nanno @quantenzitrone @erqan @igorSilCar @carrot-cookie @dircsem @tbilles @matthiakl @Moini @AlexandraAlter

### CODE REVIEW

**Comment by quantenzitrone** [comment]
I'm ok with that change, but why MIT? Why not fully public domain them with CC0, WTFPL or something?

**Comment by Julow** [comment]
I though MIT was very open and I didn't want to relicense contributors' work under more exotic licenses.

CC0 might also be a good choice. What advantage does it have over MIT in this case ?

**Comment by quantenzitrone** [comment]
https://en.wikipedia.org/wiki/Public_domain
https://en.wikipedia.org/wiki/Public-domain-equivalent_license
https://en.wikipedia.org/wiki/Permissive_software_license#Comparison_to_public_domain

**Comment by quantenzitrone** [comment]
There is no real "advantage". More of an ideology of knowledge being no one's property.

**Comment by lrvideckis** [comment]
my understanding:

MIT: you can use/fork this code as long as you keep the MIT license
unchanged (seems similar to gpl v3 license)

CC0: you can use/fork this code (under no restrictions)

As for my layout, you can use any license (preferably the more open/less
restrictions the better), thank you for asking.

**Comment by Julow** [comment]
Thanks for the information ! It seems that the Unlicense is the most appropriate as it's made for software, has the warranty clause and is the only public domain license listed on <https://choosealicense.com/licenses/>. CC0 should also work on data files but it's less clear.

A public domain license makes total sense in this case, though I'll wait to see whether a contributor complains as this removes the implicit copyright.

**Comment by Lixquid** [comment]
To add my perspective as to why this change is benificial:

I maintain the web tool for editing layout files; previously, the only example
available was an out-of-date QWERTY layout. To help more accurately reflect
layout files as they're used now, I've implemented a feature to directly read
the layout files using the same importing mechanism that anyone else uses.

The problem is, the tool is licensed under MIT, which is incompatible with the
GPL license, as it will bundle the files directly to import them.

The feature to support this is complete, and hopefully if we can come to an
agreement, I can release it soon.

I originally suggested the MIT license as it's the most used "open" license,
but I think public domain-alike licenses are a good fit as well.

**Comment by quantenzitrone** [comment]
> my understanding: MIT: you can use/fork this code as long as you keep the MIT license unchanged (seems similar to gpl v3 license)

GPL is copyleft protecting the freedom of the knowledge/software,
MIT is not protecting anything except the preservation of copyright.

**Comment by lyubomirv** [comment]
For the layout I've created, you can apply MIT, CC0 or whatever other free for use license you prefer. The less restrictions for the users, the better.

**Comment by matthiakl** [comment]
I'd rather keep a copyright notice, so preferably stick with MIT as initially suggested.

**Comment by RamKromberg** [comment]
> Thanks for the information ! It seems that the Unlicense is the most appropriate as it's made for software, has the warranty clause and is the only public domain license listed on https://choosealicense.com/licenses/. CC0 should also work on data files but it's less clear.

> A public domain license makes total sense in this case, though I'll wait to see whether a contributor complains as this removes the implicit copyright.

TL;DR Avoid Unlicense. Use 0BSD or CC0.

Keeping in mind I'm not a lawyer, the long version is that copyrights are inalienable rights in the EU so dedications to the public domain might or might not be invalidated as illegal waivers returning all rights to the original author: https://chrismorgan.info/blog/unlicense/

Having said that, I suppose if you really want to use Unlicense you could just dual-license. That is, since you want it permissive as possible, there's no harm requiring layout to be contributed under both Unlicense and CC0/0BSD.

**Comment by Julow** [comment]
Thanks for the information! I'll switch to CC0 as FSF and other groups recommend it instead of Unlicense.

I'll wait a bit more for contributors to react before merging this. 

@lyubomirv Your layout is excluded from this change. I added a copyright notice to make that clear.

**Comment by dircsem** [comment]
I Thought that I'd made a comment here last week :S 
For me whenever you choose as a license is ok, the only restriction is that it should still open source

**Comment by lyubomirv** [comment]
@Julow Why is my layout excluded? I don't need any special treatment for it and confusion, so please make it use the same license as all the others.

**Comment by Julow** [comment]
@lyubomirv Sorry! I meant to make an exception for @matthiakl and switched the names by mistake.

**Comment by Moini** [comment]
@Julow Hi, and thanks for asking! I'd like my changes to remain under a copyleft license, making it necessary for others to give attribution and to hand out the source when asked.

**Comment by Lixquid** [comment]
@Julow

Since someone's marked disagreement and we're going to have differing licenses applying to the layouts, if possible, can licenses that would not be suitable for inclusion in an MIT-licensed project (GPL, etc.) include something easily parsable in the comment header to mark it for exclusion from automated tooling?

**Comment by Julow** [comment]
@Moini The license is now updated.

@Lixquid Non-public-domain layouts will contain an XML comment near the top starting with the word "copyright". It is a requirement for new non-public-domain layouts and is added to the existing ones.
CC0-licensed layouts cannot have a copyright line and I do not plan to allow layout authors to choose a license other than the two currently in use.

**Comment by Julow** [comment]
Thanks you all for your feedback, let's merge this.

If anyone wants to exclude their layout from the new license, please open a new issue soon.
